blocks/library/latest-posts/block.js

```diff
@@ -64,7 +65,7 @@ class LatestPostsBlock extends Component {
 					checked={ displayPostDate }
 					onChange={ this.toggleDisplayPostDate }
 				/>
-				{ layout === 'grid' &&
```

Comment:
I think there's still value in rendering this control even if we have no posts. We just need to set the `MAX_POSTS_COLUMNS` const as a maximum and avoid using `.length`. What do you think?

Reply:
I considered that and decided on this approach because it seemed less fragile to me. I pictured there might be component update race conditions and that columns value might change unexpectedly while changing the number of posts, but truth be told I didn't test either hypothesis before implementing this solution. For posterity, I just tested it (I changed the max prop to `max={ Math.min( MAX_POSTS_COLUMNS, hasPosts ? latestPosts.length : columns ) }`) and the behavior gets very bizarre, even more than I was expecting. See below gif.

![columns-funky](https://user-images.githubusercontent.com/465154/34571137-ba60f1aa-f13b-11e7-9652-aba8b05a2b9f.gif)

If you have any better ideas than what I tested, I'm happy to try it out!

Reply:
Did you try something like: `max={ ! hasPosts ? MAX_POSTS_COLUMNS : Math.min( MAX_POSTS_COLUMNS, latestPosts.length ) }`

Reply:
I tried that as well, same result more or less.

Reply:
In my testing it works well, are you sure you applied this properly? Which browser are you using?

Reply:
Using Chrome/Mac, although after putting the solution I offered through the same testing behavior, it doesn't respond much differently except that it also briefly disappears.

What about sticking with a static max instead of using the posts limit? While I'm sure we all agree that it doesn't seem to make much sense to set, say, 4 columns if you only have 3 posts, it does create a smoother UI as you change the number of posts.